# Lab: Building a Reproducible Analysis Workflow

Choose your own documented dataset, develop three distinct research questions, and answer each using the six-step workflow from the Pokémon lecture. You may reuse your visual thinking dataset, but develop new questions rather than repeat its single-feature charts.


## Learning goals

- Develop three questions that can be answered with your chosen data.
- Inspect and document a dataset using its data dictionary.
- Select, group, summarize, or calculate features with Pandas.
- Connect displayed evidence, contextual sanity checks, conclusions, and limitations.


## The reproducible analysis workflow

1. **Question:** What do we want to learn?
2. **Data:** Which observations and features can help answer it?
3. **Evidence:** What should we calculate and display to answer the question?
4. **Check:** Does the result make sense in context? Do a few original records support the meaning of the calculation?
5. **Conclusion:** What claim is supported?
6. **Limitation:** What should we avoid concluding?


## Using AI tools responsibly

You may use AI to ask questions, debug code, or receive feedback. You remain responsible for understanding the work and checking suggestions. Disclose whether you used AI in the final reflection.


## Find and document your dataset

Choose your own documented CSV dataset from [Kaggle](https://www.kaggle.com/datasets) or another reputable source. You may reuse the dataset from the visual thinking lab. Choose data with:

- a clearly identified publisher or creator;
- a data dictionary or documentation explaining observations and features;
- at least one categorical and one quantitative feature;
- enough information to answer three distinct questions involving calculations or comparisons;
- content appropriate to share in class, without private or personally identifying information; and
- a manageable file size.

Save the CSV in this notebook's `data` folder. Use a descriptive DataFrame name ending in `_df` throughout your analysis.


**What is the dataset's title, publisher or creator, direct source URL, and data dictionary URL or location?**

Your answer: The dataset is "Full Pokemons and Moves Datasets", created by Thiago Amancio and published on Kaggle. I used the `metadata_pokemon.csv` file.

- Source: [Full Pokemons and Moves Datasets on Kaggle](https://www.kaggle.com/datasets/thiagoamancio/full-pokemons-and-moves-datasets)
- Data dictionary: the "About Dataset" section of the same Kaggle page, under "Pokémon Dataset Key Columns"


**Briefly describe the dataset and relevant features using its documentation. What does one row represent? Include units and the meaning of blank values where documented.**

Your answer: Each row represents one Pokémon species, identified by its National Pokédex number (`id`), from #1 (Bulbasaur) to #998 (Baxcalibur). The file has 998 rows and 12 features. Every `id` and `name` is unique, so each species appears only once. For Pokémon with more than one form, only one form is listed (for example, `Deoxys-Normal` and `Tornadus-Incarnate`), and other forms are not included.

Relevant features:
- `hp`, `attack`, `defense`, `special_attack`, `special_defense`, `speed`: the Pokémon's base stats. These are game values with no unit.
- `height`: listed in decimeters (Bulbasaur's 7 = 0.7 m).
- `weight`: the dictionary says grams, but the values match hectograms (tenths of a kilogram). Bulbasaur's 69 matches its real weight of 6.9 kg, not 69 g.
- `type_1`: the Pokémon's primary type (18 types).
- `type_2`: the Pokémon's secondary type, "if applicable" according to the dictionary. A blank means the Pokémon has only one type, not that the value is unknown, so those rows are kept.


## Plan three investigations

Develop three distinct research questions about your dataset. Each must require a calculation or comparison, not simply looking up a value. Consider grouped averages, percentages meeting a condition, or comparisons between features, as in the lecture; adapt the questions to what your data means. Complete all six steps for each question. Record each prediction before running its calculations.


## Investigation 1

### Step 1: Question

State a specific question that your dataset can answer. Define any important groups or comparisons.


**What is your research question 1?**

Your answer: Which primary type (`type_1`) of Pokémon has the highest average Speed? I will compare the mean `speed` of all 18 primary types, using all 998 Pokémon.


### Prediction — before calculating


**What do you predict the result will be, and why?**

Your answer: I think Electric will have the highest average Speed, because a lot of Electric-type Pokémon are themed around being fast, like lightning or electricity.


### Step 2: Data


Import Pandas and load your CSV into a descriptively named DataFrame ending in `_df`. Inspect the data once here, using separate cells for each operation.


In [1]:
import pandas as pd

In [2]:
pokemon_df = pd.read_csv("data/metadata_pokemon.csv")

In [3]:
pokemon_df.head()

,name,id,hp,attack,defense,special_attack,special_defense,speed,height,weight,type_1,type_2
0,Bulbasaur,1,45,49,49,65,65,45,7,69,Grass,Poison
1,Ivysaur,2,60,62,63,80,80,60,10,130,Grass,Poison
2,Venusaur,3,80,82,83,100,100,80,20,1000,Grass,Poison
3,Charmander,4,39,52,43,60,50,65,6,85,Fire,NaN
4,Charmeleon,5,58,64,58,80,65,80,11,190,Fire,NaN


In [4]:
pokemon_df.tail()

,name,id,hp,attack,defense,special_attack,special_defense,speed,height,weight,type_1,type_2
993,Iron-Moth,994,80,70,60,140,110,110,12,360,Fire,Poison
994,Iron-Thorns,995,100,134,110,70,84,72,16,3030,Rock,Electric
995,Frigibax,996,65,75,45,35,45,55,5,170,Dragon,Ice
996,Arctibax,997,90,95,66,45,65,62,8,300,Dragon,Ice
997,Baxcalibur,998,115,145,92,75,86,87,21,2100,Dragon,Ice


In [5]:
pokemon_df.shape

(998, 12)

In [6]:
pokemon_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 998 entries, 0 to 997
Data columns (total 12 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   name             998 non-null    object
 1   id               998 non-null    int64 
 2   hp               998 non-null    int64 
 3   attack           998 non-null    int64 
 4   defense          998 non-null    int64 
 5   special_attack   998 non-null    int64 
 6   special_defense  998 non-null    int64 
 7   speed            998 non-null    int64 
 8   height           998 non-null    int64 
 9   weight           998 non-null    int64 
 10  type_1           998 non-null    object
 11  type_2           502 non-null    object
dtypes: int64(9), object(3)
memory usage: 93.7+ KB


In [7]:
pokemon_df.isnull().sum()

name                 0
id                   0
hp                   0
attack               0
defense              0
special_attack       0
special_defense      0
speed                0
height               0
weight               0
type_1               0
type_2             496
dtype: int64

**Which observations and features will you use? Explain relevant units, any filtering, and how you will handle missing or not-applicable values using the dictionary. For a percentage, define the denominator.**

Your answer: I will use all 998 Pokémon with no filtering. The features are `type_1` (primary type, which sorts the Pokémon into 18 groups) and `speed` (a base stat with no unit). Neither feature has any missing values, so every Pokémon is included. I am not using `type_2`, so its blanks don't affect this question. This question uses an average, not a percentage, so there is no denominator to define.


### Step 3: Evidence

Write the code that answers your question. Break the analysis into separate cells for individual operations, as in the lecture, and add cells as needed. Display intermediate results where useful and the final result that answers the question. Explain any new calculated features. You may display a grouped summary directly without assigning it to a variable.


In [8]:
pokemon_df.groupby("type_1")["speed"].agg(["count", "mean"])

,count,mean
type_1,,
Bug,83,61.265060
Dark,41,74.463415
Dragon,36,79.111111
Electric,57,84.368421
Fairy,28,63.178571
Fighting,39,68.307692
Fire,65,73.846154
Flying,9,83.000000
Ghost,34,61.823529


In [9]:
pokemon_df.groupby("type_1")["speed"].agg(["count", "mean"]).sort_values("mean", ascending=False)

,count,mean
type_1,,
Electric,57,84.368421
Flying,9,83.000000
Dragon,36,79.111111
Dark,41,74.463415
Fire,65,73.846154
Psychic,60,72.083333
Fighting,39,68.307692
Normal,117,68.051282
Poison,38,67.973684


**Which specific value or comparison answers your question? Explain how your calculations connect the selected data to that result.**

Your answer: Electric has the highest mean Speed at about 84.37, based on 57 Electric-type Pokémon. Flying is a close second at 83.00 and Dragon is third at about 79.11. Rock has the lowest at about 53.42. I grouped all 998 Pokémon by `type_1`, then used `.agg(["count", "mean"])` to calculate how many Pokémon are in each type and their average `speed`. Sorting by `mean` from largest to smallest put the type with the highest average Speed at the top.


### Step 4: Check

Perform a sanity check tied to the meaning of your question. Explain what values or patterns would make you suspicious and whether your result makes sense. Depending on the analysis, consider plausible values and units, whether indicators match original labels, or whether the ranges within comparison groups match their definitions. Use additional code only when it helps this check; do not simply repeat the initial row count or missing-value inspection. An unexpected result is a reason to investigate, not to change the result to fit your prediction.


In [10]:
pokemon_df.groupby("type_1")["speed"].agg(["min", "max"])

,min,max
type_1,,
Bug,5,160
Dark,35,125
Dragon,40,142
Electric,15,200
Fairy,15,138
Fighting,25,138
Fire,20,126
Flying,55,123
Ghost,15,135


In [11]:
pokemon_df.loc[pokemon_df["type_1"] == "Flying", ["name", "type_1", "type_2", "speed"]]

,name,type_1,type_2,speed
640,Tornadus-Incarnate,Flying,NaN,111
713,Noibat,Flying,Dragon,55
714,Noivern,Flying,Dragon,123
820,Rookidee,Flying,NaN,57
821,Corvisquire,Flying,NaN,77
822,Corviknight,Flying,Steel,67
844,Cramorant,Flying,Water,85
961,Bombirdier,Flying,Dark,82
972,Flamigo,Flying,Fighting,90


In [12]:
pokemon_df.loc[(pokemon_df["type_1"] == "Electric") & (pokemon_df["name"] != "Regieleki"), "speed"].mean()

np.float64(82.30357142857143)

**What would make you suspicious, and why? Does your evidence make sense in context? Describe any issue you found and how you addressed it.**

Your answer: I would be suspicious of any negative Speed values, a type's minimum or maximum outside the dataset's range of 5 to 200, or a group mean outside its own min and max. None of those showed up, so the values make sense. Two things stood out. First, Flying only has 9 Pokémon as a primary type. Listing them shows that this isn't an error: most bird Pokémon (like Pidgey) are Normal/Flying, so Flying is usually their secondary type. Their speeds range from 55 to 123, so no single Pokémon is pulling the Flying average up. Second, Electric's max Speed is 200, which belongs to Regieleki. Without Regieleki, Electric's mean would be about 82.30, slightly below Flying's 83.00. I kept Regieleki because it's in the data, but it shows that the top two types are very close.


### Step 5: Conclusion


**Answer your research question in a sentence using the numerical evidence. How does it compare with your prediction?**

Your answer: Electric has the highest average Speed of any primary type at about 84.37 across 57 Pokémon, just ahead of Flying (83.00, 9 Pokémon) and Dragon (about 79.11, 36 Pokémon). This matches my prediction that Electric would be the fastest, although its lead over Flying is small.


### Step 6: Limitation


**What should a reader avoid concluding from this result, and why? Connect your limitation to this question and dataset.**

Your answer: A reader should not conclude that every Electric-type Pokémon is faster than every Pokémon of another type. Electric's speeds range from 15 to 200, so plenty of Electric types are slower than Pokémon of other types. The result also only groups by primary type (`type_1`), so a Pokémon like Pidgey (Normal/Flying) counts as Normal, not Flying. That is why Flying only has 9 Pokémon, and an average from 9 Pokémon can change a lot if even one is added or removed. Also, Electric's lead is small and depends on Regieleki, so the ranking between Electric and Flying could easily flip in a different version of the data.


## Investigation 2

### Step 1: Question

State a specific question that your dataset can answer. Define any important groups or comparisons.


**What is your research question 2?**

Your answer: What percent of Pokémon weigh more than 100 kg? I will convert `weight` from hectograms to kilograms and count the Pokémon over 100 kg (not including exactly 100 kg), out of all 998 Pokémon.


### Prediction — before calculating


**What do you predict the result will be, and why?**

Your answer: I think around 12% of Pokémon will weigh more than 100 kg. Legendary Pokémon tend to be heavy, and quite a few non-legendary Pokémon are big too, like Metagross and Snorlax. Some small Pokémon are also very heavy, like Cosmoem, so size alone doesn't decide it.


### Step 2: Data


Reuse the DataFrame and initial inspection from Investigation 1. Identify the observations and features relevant to this question; inspect additional features only if needed.


**Which observations and features will you use? Explain relevant units, any filtering, and how you will handle missing or not-applicable values using the dictionary. For a percentage, define the denominator.**

Your answer: I will use all 998 Pokémon with no filtering, and the `name` and `weight` features. The dictionary says `weight` is in grams, but the values match hectograms (Bulbasaur's 69 = 6.9 kg), so I will divide by 10 to convert to kilograms. `weight` has no missing values, so every Pokémon is included. The denominator is all 998 Pokémon, and the numerator is the number of Pokémon weighing more than 100 kg. A Pokémon at exactly 100 kg does not count.


### Step 3: Evidence

Write the code that answers your question. Break the analysis into separate cells for individual operations, as in the lecture, and add cells as needed. Display intermediate results where useful and the final result that answers the question. Explain any new calculated features. You may display a grouped summary directly without assigning it to a variable.


In [13]:
pokemon_df["weight_kg"] = pokemon_df["weight"] / 10
pokemon_df[["name", "weight", "weight_kg"]].head()

,name,weight,weight_kg
0,Bulbasaur,69,6.9
1,Ivysaur,130,13.0
2,Venusaur,1000,100.0
3,Charmander,85,8.5
4,Charmeleon,190,19.0


In [14]:
pokemon_df["over_100_kg"] = (pokemon_df["weight_kg"] > 100).astype(int)
pokemon_df[["name", "weight_kg", "over_100_kg"]].head(10)

,name,weight_kg,over_100_kg
0,Bulbasaur,6.9,0
1,Ivysaur,13.0,0
2,Venusaur,100.0,0
3,Charmander,8.5,0
4,Charmeleon,19.0,0
5,Charizard,90.5,0
6,Squirtle,9.0,0
7,Wartortle,22.5,0
8,Blastoise,85.5,0
9,Caterpie,2.9,0


In [15]:
over_100_kg_count = pokemon_df["over_100_kg"].sum()
over_100_kg_count

np.int64(163)

In [16]:
over_100_kg_percent = over_100_kg_count / len(pokemon_df["over_100_kg"]) * 100
over_100_kg_percent

np.float64(16.33266533066132)

In [17]:
print("Over 100 kg:", over_100_kg_count)

Over 100 kg: 163


In [18]:
print("All Pokémon:", len(pokemon_df["over_100_kg"]))

All Pokémon: 998


In [19]:
print("Percent:", round(over_100_kg_percent, 2))

Percent: 16.33


**Which specific value or comparison answers your question? Explain how your calculations connect the selected data to that result.**

Your answer: 163 out of 998 Pokémon weigh more than 100 kg, which is about 16.33%. I divided `weight` by 10 to convert hectograms to kilograms and saved it as `weight_kg`. Then I made an indicator, `over_100_kg`, that is 1 when `weight_kg` is greater than 100 and 0 otherwise. Summing the indicator gives the count of Pokémon over 100 kg (163), and dividing by all 998 Pokémon and multiplying by 100 gives the percentage.


### Step 4: Check

Perform a sanity check tied to the meaning of your question. Explain what values or patterns would make you suspicious and whether your result makes sense. Depending on the analysis, consider plausible values and units, whether indicators match original labels, or whether the ranges within comparison groups match their definitions. Use additional code only when it helps this check; do not simply repeat the initial row count or missing-value inspection. An unexpected result is a reason to investigate, not to change the result to fit your prediction.


In [20]:
pokemon_df.loc[pokemon_df["name"].isin(["Pikachu", "Snorlax", "Metagross", "Cosmoem"]), ["name", "weight", "weight_kg", "over_100_kg"]]

,name,weight,weight_kg,over_100_kg
24,Pikachu,60,6.0,0
142,Snorlax,4600,460.0,1
375,Metagross,5500,550.0,1
789,Cosmoem,9999,999.9,1


In [21]:
pokemon_df.groupby("over_100_kg")["weight_kg"].agg(["count", "min", "max"])

,count,min,max
over_100_kg,,,
0,835,0.1,100.0
1,163,100.5,999.9


**What would make you suspicious, and why? Does your evidence make sense in context? Describe any issue you found and how you addressed it.**

Your answer: I would be suspicious if known Pokémon had weights 10 times too big or too small, which would mean the unit conversion was wrong, or if any Pokémon at or below 100 kg was marked as over 100 kg. Pikachu (6.0 kg), Snorlax (460.0 kg), Metagross (550.0 kg), and Cosmoem (999.9 kg) all match their real Pokédex weights, which confirms that `weight` is in hectograms and the conversion is correct. If the values were really grams like the dictionary says, Snorlax would only weigh 4.6 kg. The indicator groups also match their meaning: the over-100 group's minimum is 100.5 kg, the other group's maximum is exactly 100.0 kg, and the counts (835 + 163) add up to all 998 Pokémon. The only issue I found was the dictionary's unit, which I handled by dividing by 10.


### Step 5: Conclusion


**Answer your research question in a sentence using the numerical evidence. How does it compare with your prediction?**

Your answer: About 16.33% of Pokémon (163 out of 998) weigh more than 100 kg. That is higher than my prediction of around 12%, so there are more heavy Pokémon than I expected, but my guess was still in the right range.


### Step 6: Limitation


**What should a reader avoid concluding from this result, and why? Connect your limitation to this question and dataset.**

Your answer: A reader should not assume that heavy Pokémon are also big. Cosmoem weighs 999.9 kg but is tiny, so this result says nothing about size or height. A reader also shouldn't assume the heavy Pokémon are mostly legendaries, because this dataset has no legendary column, so I can't tell how many of the 163 are legendary. The 100 kg cutoff is also a choice I made: Venusaur is exactly 100.0 kg and isn't counted, so a slightly different cutoff would give a different percentage. Finally, this is the percent of Pokémon species, not the percent of Pokémon a player would actually run into in a game.


## Investigation 3

### Step 1: Question

State a specific question that your dataset can answer. Define any important groups or comparisons.


**What is your research question 3?**

Your answer: Do dual-type Pokémon have a higher average base stat total than single-type Pokémon? I will add up each Pokémon's six base stats (`hp`, `attack`, `defense`, `special_attack`, `special_defense`, `speed`) into a new `base_stat_total` feature, then compare the mean total of Pokémon with a secondary type (`type_2` not blank) to those without one (`type_2` blank).


### Prediction — before calculating


**What do you predict the result will be, and why?**

Your answer: I think dual-type Pokémon will have a higher average base stat total. I think more fully evolved Pokémon have two types than one, and fully evolved Pokémon have higher stat totals on average.


### Step 2: Data


Reuse the DataFrame and initial inspection from Investigation 1. Identify the observations and features relevant to this question; inspect additional features only if needed.


**Which observations and features will you use? Explain relevant units, any filtering, and how you will handle missing or not-applicable values using the dictionary. For a percentage, define the denominator.**

Your answer: I will use all 998 Pokémon with no filtering. The features are the six base stats (`hp`, `attack`, `defense`, `special_attack`, `special_defense`, `speed`), which are game values with no unit, and `type_2`. None of the base stats are missing. `type_2` has 496 blanks, but the dictionary says a secondary type is only listed "if applicable," so a blank means the Pokémon has only one type, not that the value is unknown. I will keep those rows as the single-type group instead of dropping them. This question compares averages, not a percentage, so there is no denominator to define.


### Step 3: Evidence

Write the code that answers your question. Break the analysis into separate cells for individual operations, as in the lecture, and add cells as needed. Display intermediate results where useful and the final result that answers the question. Explain any new calculated features. You may display a grouped summary directly without assigning it to a variable.


In [22]:
pokemon_df["base_stat_total"] = pokemon_df["hp"] + pokemon_df["attack"] + pokemon_df["defense"] + pokemon_df["special_attack"] + pokemon_df["special_defense"] + pokemon_df["speed"]
pokemon_df[["name", "hp", "attack", "defense", "special_attack", "special_defense", "speed", "base_stat_total"]].head()

,name,hp,attack,defense,special_attack,special_defense,speed,base_stat_total
0,Bulbasaur,45,49,49,65,65,45,318
1,Ivysaur,60,62,63,80,80,60,405
2,Venusaur,80,82,83,100,100,80,525
3,Charmander,39,52,43,60,50,65,309
4,Charmeleon,58,64,58,80,65,80,405


In [23]:
pokemon_df["is_dual_type"] = pokemon_df["type_2"].notna().astype(int)
pokemon_df[["name", "type_1", "type_2", "is_dual_type"]].head(10)

,name,type_1,type_2,is_dual_type
0,Bulbasaur,Grass,Poison,1
1,Ivysaur,Grass,Poison,1
2,Venusaur,Grass,Poison,1
3,Charmander,Fire,NaN,0
4,Charmeleon,Fire,NaN,0
5,Charizard,Fire,Flying,1
6,Squirtle,Water,NaN,0
7,Wartortle,Water,NaN,0
8,Blastoise,Water,NaN,0
9,Caterpie,Bug,NaN,0


In [24]:
pokemon_df.groupby("is_dual_type")["base_stat_total"].agg(["count", "mean"])

,count,mean
is_dual_type,,
0,496,402.899194
1,502,445.507968


**Which specific value or comparison answers your question? Explain how your calculations connect the selected data to that result.**

Your answer: Dual-type Pokémon have a mean base stat total of about 445.51 (502 Pokémon), compared with about 402.90 for single-type Pokémon (496 Pokémon), a difference of about 42.61 points. I created `base_stat_total` by adding each Pokémon's six base stats together. Then I made an indicator, `is_dual_type`, that is 1 when `type_2` has a value and 0 when it is blank. Grouping by `is_dual_type` and using `.agg(["count", "mean"])` gave the number of Pokémon and the average base stat total in each group.


### Step 4: Check

Perform a sanity check tied to the meaning of your question. Explain what values or patterns would make you suspicious and whether your result makes sense. Depending on the analysis, consider plausible values and units, whether indicators match original labels, or whether the ranges within comparison groups match their definitions. Use additional code only when it helps this check; do not simply repeat the initial row count or missing-value inspection. An unexpected result is a reason to investigate, not to change the result to fit your prediction.


In [25]:
pokemon_df.loc[pokemon_df["name"].isin(["Bulbasaur", "Pikachu", "Mewtwo"]), ["name", "type_1", "type_2", "is_dual_type", "base_stat_total"]]

,name,type_1,type_2,is_dual_type,base_stat_total
0,Bulbasaur,Grass,Poison,1,318
24,Pikachu,Electric,NaN,0,320
149,Mewtwo,Psychic,NaN,0,680


In [26]:
pokemon_df.groupby("is_dual_type")["base_stat_total"].agg(["min", "median", "max"])

,min,median,max
is_dual_type,,,
0,175,409.5,720
1,185,475.0,690


**What would make you suspicious, and why? Does your evidence make sense in context? Describe any issue you found and how you addressed it.**

Your answer: I would be suspicious if `is_dual_type` didn't match the original `type_2` labels, or if any base stat total fell outside the real range of about 175 to 720. Bulbasaur (Grass/Poison) is correctly marked 1 with a total of 318, and Pikachu (Electric) and Mewtwo (Psychic) are correctly marked 0 with totals of 320 and 680. Both groups stay within the expected range (175 to 720 for single-type, 185 to 690 for dual-type), and the group counts (496 + 502) add up to all 998 Pokémon, with 496 matching the number of blank `type_2` values. The medians (409.5 vs. 475.0) show the same pattern as the means, so a few extreme Pokémon aren't creating the difference. The highest total in the data (720) is actually in the single-type group. I did not find any issues.


### Step 5: Conclusion


**Answer your research question in a sentence using the numerical evidence. How does it compare with your prediction?**

Your answer: Yes, dual-type Pokémon have a higher average base stat total (about 445.51 across 502 Pokémon) than single-type Pokémon (about 402.90 across 496 Pokémon), a difference of about 42.61 points. This matches my prediction that dual-type Pokémon would average higher.


### Step 6: Limitation


**What should a reader avoid concluding from this result, and why? Connect your limitation to this question and dataset.**

Your answer: A reader should not conclude that having a second type causes a Pokémon to have higher stats. My prediction was based on fully evolved Pokémon being more likely to have two types. While I can remember which Pokémon are fully evolved, that isn't recorded anywhere in this dataset, and it has no evolution stage or legendary column, so I can't check whether that explains the difference. A reader also shouldn't assume every dual-type Pokémon is stronger than every single-type Pokémon. The ranges overlap a lot, and the highest total in the whole dataset (720) belongs to a single-type Pokémon. Finally, base stat total doesn't include abilities, moves, or type matchups, so a higher total doesn't mean a Pokémon is better in battle.


## AI-use reflection


**State whether you used an AI tool. If you did, describe one suggestion you checked and what you accepted, changed, or rejected. If you did not, state that you did not use one.**

Your answer: Yes, I used Claude. One suggestion I checked was to calculate Electric's average Speed without Regieleki, since its Speed of 200 was the highest in the Electric group. I ran it and got about 82.30, which is below Flying's 83.00. I accepted that this shows Electric's lead depends on one Pokémon, but I did not remove Regieleki from my analysis, because it is a real Pokémon in the data and removing it would just be changing the result.


## Submission checklist

- Include your chosen dataset, source, and dictionary reference.
- Complete three distinct questions, each with a prediction and all six workflow steps.
- Use a descriptive DataFrame name ending in `_df` and relative file paths.
- Restart the kernel and run all cells in order; verify that numerical claims match displayed output.
- Include a contextual sanity check, supported conclusion, and relevant limitation for each question.
- Complete every written response and the AI-use reflection.
